install.sh

```diff
@@ -457,151 +456,69 @@ Mac system, you can use Vagrant or another virtualization tool. See our install
   fi
 }
 
-detect_how_to_enable_userns_or_die_trying() {
-  # You might think it's a little bit silly to embed an x86_64 binary
-  # in this shell script, just find out if user namespaces work for
-  # unprivileged users.
-  #
-  # However, here's the story:
-  #
-  # - Many people use the Debian backports kernel (where user namespaces work great) with older
-  #   userspace, so we can't run unshare(1) with the --user option to test it, since their version
-  #   of unshare(1) doesn't have a --user option.
-  #
-  # - Many people run Arch Linux, where user namespaces are disabled as a kernel option.
-  #
-  # - Many people run Debian and/or Ubuntu, where user namespaces are enabled but require a sysctl
-  #   to enable.
-  #
-  # - We used to compile a test binary, but sometimes people would install Sandstorm on cloud VMs
-  #   where there is no compiler.
-  #
-  # We export the following variable:
-  #
-  # - USERNS_CLONE_UNPRIVILEGED_NEEDS_SYSCTL_SET = yes/no
-  #
-  # Also at the end of the function, if the program is still running, unprivileged userns either
-  # works or can be made to work.
-
-  USERNS_CLONE_UNPRIVILEGED_NEEDS_SYSCTL_SET=no
-
-  # Let's run a test program that creates a user namespace. If it passes, we know that no further
-  # configuration is required.
-  #
-  # Source of this program: https://github.com/sandstorm-io/check-for-unprivileged-userns
-  local USERNS_TEST_PROGRAM="$(mktemp)"
-  printf '\x7fELF\x02\x01\x01\x00kmc!!!\n\x00\x02\x00>\x00\x01\x00\x00\x00x\x00@\x00\x00\x00\x00\x00@\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00@\x008\x00\x01\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x07\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00@\x00\x00\x00\x00\x00\x00\x00@\x00\x00\x00\x00\x00\x9d\x00\x00\x00\x00\x00\x00\x00\x9d\x00\x00\x00\x00\x00\x00\x00\x00\x10\x00\x00\x00\x00\x00\x00\xb8i\x00\x00\x00\xbf\xfe\xff\x00\x00\x0f\x05\xb8\x10\x01\x00\x00\xbf\x00\x00\x00\x10\x0f\x05H\x89\xc1\xb8<\x00\x00\x00H\x89\xcf\x0f\x05' > "$USERNS_TEST_PROGRAM"
-  chmod a+x "$USERNS_TEST_PROGRAM"
-  if "$USERNS_TEST_PROGRAM" ; then
-    return  # success!
-  fi
-
-  # Clean up after ourselves.
-  rm -f "$USERNS_TEST_PROGRAM"
-
-  # If we're still here, the test program failed.
+maybe_enable_userns_sysctl() {
+  # This function enables the Debian/Ubuntu-specific unprivileged
+  # userns sysctl, if the system has it and we want it.
 
-  # Let's see if there's a Debian-style sysctl. If not, bail out because we have no idea how to help
-  # the user.
-  if [ ! -e /proc/sys/kernel/unprivileged_userns_clone ]; then
-    fail "E_NO_USERNS" "Your kernel does not appear to be compiled with" \
-         "support for unprivileged user namespaces (CONFIG_USER_NS=y), or something else is" \
-         "preventing creation of user namespaces. This feature is critical for sandboxing." \
-         "Read more at: " \
-         "https://docs.sandstorm.io/en/latest/administering/faq/#enabling-user-namespaces"
```

Comment:
Notably: In https://github.com/sandstorm-io/sandstorm/pull/2694 I plan to remove this section from the docs. Happily, install.sh is also about to stop referring to it.

This does mean that old copies of install.sh will link here. I keep wondering if we should avoid breaking that link. But I guess people will be OK.